pkg/volume/csi/csi_attacher.go

```diff
@@ -269,6 +269,25 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 		return err
 	}
 
+	dataDir := filepath.Dir(deviceMountPath)
+	if err := os.MkdirAll(dataDir, 0750); err != nil {
+		glog.Error(log("attacher.MountDevice failed to create dir %#v:  %v", dataDir, err))
+		return err
+	}
+
+	data := map[string]string{
+		volDataKey.volHandle:  csiSource.VolumeHandle,
+		volDataKey.driverName: csiSource.Driver,
+	}
+	if err := saveVolumeData(dataDir, volDataFileName, data); err != nil {
```

Comment:
@vladimirvivien: you recently moved `saveVolumeData(...)` from `SetupAt(...)` to `NewMounter(...)`. Does something similar need to happen here?

Reply:
@saad-ali unfortunately, the way the Attacher API is written, no params is available during `csi_plugin.go#NewAttacher/#NewDetacher` that would allow the volume info to be persisted at that point as is done wit NewMounter/Unmounter.  So, the saving of the volume info has to happen during `MountDevice` call.